include/openssl/dh.h

```diff
@@ -112,7 +112,7 @@ DECLARE_ASN1_ITEM(DHparams)
 # define i2d_DHxparams_bio(bp,x) \
     ASN1_i2d_bio_of_const(DH, i2d_DHxparams, bp, x)
 
-DH *DHparams_dup(DH *);
+DECLARE_ASN1_DUP_FUNCTION_name(DH, DHparams)
```

Comment:
> 
> 
> I'm going to assume that you scripted an automatic replacement of every `_dup` declaration with the corresponding `DECLARE_ASN1_DUP_FUNCTION` line. That script obviously went rogue, as it affected non-ASN.1 types. Or, if it was intentional, I'm just gonna say "no", because this is just deeply confusing (I'll go as far as saying that this is an abuse of that macro)

The process I used was semi-automatic. I tried avoiding the use of `DECLARE_ASN1_DUP_FUNCTION` for non-ASN.1 types, but overlooked two. See my individual responses to your many 'no' comments.

Reply:
`DHparams` does look like an ASN.1 type to me. Or why is there for instance
```
include/openssl/dh.h:DECLARE_ASN1_ITEM(DHparams)
```

Reply:
Good point.